### PR TITLE
using password instead of pin

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: freebasics/settings/base.py [hunk 1]
```diff
@@ -181,6 +181,8 @@
     'django.core.context_processors.request',
     'molo.core.context_processors.locale',
     'molo.profiles.context_processors.get_profile_data',
+    'freebasics.context_processors.default_forms',
```
Comment (miltontony) [inline freebasics/settings/base.py]: This needs to go

File: freebasics/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def setUp(self):
+        self.client = Client()
+
+    def test_register_view_invalid_form(self):
+        # NOTE: empty form submission
+        response = self.client.post(reverse('user_register'), {
+            'username': 'testing',
+            'password': 'testing',
+        })
+        print response
```
Comment (miltontony) [inline freebasics/tests/test_views.py]: stray pring

File: freebasics/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+from django.test import TestCase, Client
+
+
+class RegistrationViewTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+
+    def test_register_view_invalid_form(self):
+        # NOTE: empty form submission
```
Comment (miltontony) [inline freebasics/tests/test_views.py]: copy paste

File: freebasics/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        response = self.client.post(reverse('user_register'), {
+            'username': 'testing',
+            'password': 'testing',
+        })
+        print response
+        self.assertRedirects(response, reverse('user_register'))
```
Comment (miltontony) [inline freebasics/tests/test_views.py]: shouldn't this be redirecting to `registration_done`
Please check that it's actually not just returning an error.

**Comment by codiebeulaine** [comment]
@saeedmb8 ready for review

File: freebasics/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        self.client = Client()
+
+    def test_registration_with_password(self):
+        response = self.client.post(reverse('user_register'), {
+            'username': 'testing',
+            'password': 'testing',
+        })
+        self.assertRedirects(response, reverse(
+            'molo.profiles:user_register'))
+        self.assertEquals(User.objects.all().count(), 1)
```
Comment (miltontony) [inline freebasics/tests/test_views.py:21]: @codiebeulaine it's always good to test before and after. So assert there are 0 users before. Also, can you test that the user created is the correct username

**Comment by smarzban** [comment]
:+1: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake:

**Pull request merged**